zcutil/fetch-pyarams.py

```diff
@@ -0,0 +1,271 @@
+        buffer = p1.communicate()[0]
+    except:
+        print("ERROR in IPFS process call")
+
+    if p1.returncode != 0:
+        print("Likely got an error from subprocess")
+
+def use_http(filename):
+    use_https(filename)
+
+def use_https(filename):
+    path = PARAMS_DIR + filename + ".dl"
+    response = requests.get(PARAMS_URL + filename , stream=True)
+    size = 0
+    total_size = int(response.headers['Content-Length'])
+    block_size = 1024
+
+    logging.info("Retrieving : %s", PARAMS_URL + filename)
+    logging.info("Length : ~ %s KB [ %s ]", str(total_size /  block_size), response.headers['Content-Type'] )
+    logging.info("Saving to : %s", path)
+    logging.debug(response.headers)
+   
+    with open(path, "wb") as handle:
+        lock(handle, LOCK_EX )
+        for chunk in tqdm(response.iter_content(block_size), total= int(total_size // block_size), unit="KB", desc=filename + ".dl"):
+            if chunk:
+                handle.write(chunk)
+                size += len(chunk)
+            else:
+                logging.warning("Received response that was not content data.")
+        unlock(handle)
+    
+    logging.info(" '%s' saved [%d/%d]", filename, size, total_size)
+    
+    if size == total_size :
+        verify_file(path, PARAM_FILES.get(filename), DOWNLOADING)
+
+def download_file(filename, protocol):
+    if protocol == "HTTPS":
+        use_https(filename)
+    elif protocol == "HTTP":
+        use_http(filename)
+    elif protocol == "IPFS":
+        use_ipfs(filename)
+    else:
+        logging.error("%s is not currently supported for retrieving Zcash params.\n", protocol)
+
+def verify_file(filename, sha256, download_state):
+    logging.debug("Checking SHA256 for: %s", filename) 
+    with open(filename, 'rb') as f:
+        try:
+    	    contents = f.read()
+        except:
+            logging.exception("Unable to read in data blocks to verify SHA256 for: %s", filename)
+        
+        local_sha256 =  hashlib.sha256(contents).hexdigest()
+        if local_sha256 != sha256 :
+            logging.error("Download failed: SHA256 on %s does NOT match.", filename)
+            return
+            
+    if download_state == DOWNLOADING :
+        logging.info("Download successful!")
+        logging.info("%s: OK", filename)
+        
+        try:
+            os.rename(filename, filename[:-3])
+        except:
+            logging.exception("Unable to rename file.")
+
+        logging.info("renamed '%s' -> '%s' \n", filename, filename[:-3])
+    
+    if download_state == DOWNLOADED :
+        logging.info("%s: OK", filename)
+
+def get_params(param_file_list, protocol):
+    for filename in param_file_list:
+        download_file(filename, protocol)
+
+def check_params(param_file_list, protocol):
+    for key in param_file_list:
+        if os.path.exists(PARAMS_DIR + key) == True :
+            verify_file(PARAMS_DIR + key , param_file_list.get(key), DOWNLOADED )
```

Comment:
Since we don't provide the Sprout keys anymore, I guess this needs to be revised?